src/cplus_plugin/gui/settings/carbon_options.py

```diff
@@ -496,6 +509,65 @@ def _on_irrecoverable_carbon_layer_changed(self, layer: qgis.core.QgsMapLayer):
         if layer is not None:
             self.fw_irrecoverable_carbon.setFilePath(layer.source())
 
+    def _on_download_stored_carbon(self, checked: bool):
+        """Slot raised to initiate download of the global stored carbon
+        dataset.
+        """
+        stored_carbon_url = settings_manager.get_value(
+            Settings.STORED_CARBON_ONLINE_SOURCE, default=""
+        )
+        if not stored_carbon_url:
+            self.message_bar.pushWarning(
+                self.tr("CPLUS Stored Carbon"),
+                self.tr("URL for downloading online data is not defined"),
+            )
+            return
+
+        default_dir = settings_manager.get_value(Settings.LAST_DATA_DIR)
+        if not default_dir:
+            default_dir = settings_manager.get_value(Settings.BASE_DIR)
+
+        file_path, _ = QFileDialog.getSaveFileName(
+            self,
+            self.tr("Specify Save Location of Stored Carbon"),
+            default_dir,
+            self._tif_file_filter,
+        )
+        if not file_path:
+            return
+
+        # Update last used directory
+        settings_manager.set_value(Settings.LAST_DATA_DIR, os.path.dirname(file_path))
+
+        self.fw_biomass.setFilePath(file_path)
+
+        # Initiate download
+        stored_carbon_download_task = start_stored_carbon_download(local_path=file_path)
```

Comment:
`stored_carbon_url` is not used here. Will the functions called use `stored_carbon_url` or any default URL?